modules/renter/persist.go

```diff
@@ -270,13 +272,17 @@ func (r *Renter) shareFiles(nicknames []string, w io.Writer) error {
 		return err
 	}
 
+	// Create compressor.
+	zip, _ := gzip.NewWriterLevel(w, gzip.BestCompression)
+	defer zip.Close()
+
 	// Write each file.
 	for _, name := range nicknames {
 		file, exists := r.files[name]
 		if !exists {
 			return ErrUnknownPath
 		}
-		err := file.save(w)
+		err := file.save(zip)
```

Comment:
`save` is generic; it just translates a file to a byte sequence. It doesn't have any concept of chaining or compression.
hmm. I suppose that makes it a `SiaMarshaller`. Maybe that's the way to go.
You're right though, some extra documentation is warranted.